web-common/src/features/sources/modal/AddDataForm.svelte

```diff
@@ -208,13 +205,46 @@
     } else {
       const values =
         hasOnlyDsn() || connectionTab === "dsn" ? $dsnForm : $paramsForm;
-      return compileConnectorYAML(connector, values, {
-        fieldFilter: (property) => !property.noPrompt,
-        orderedProperties:
-          hasOnlyDsn() || connectionTab === "dsn"
-            ? filteredDsnProperties
-            : filteredParamsProperties,
-      });
+
+      let previewValues = { ...values };
+
+      if (connector.sourceProperties) {
+        for (const prop of connector.sourceProperties) {
+          if (prop.key && prop.required && !(prop.key in previewValues)) {
+            if (prop.placeholder) {
+              previewValues[prop.key] = prop.placeholder;
+            }
+          }
+        }
+      }
+
+      // Apply DuckDB rewrite logic for preview
+      const [rewrittenConnector, rewrittenFormValues] = maybeRewriteToDuckDb(
+        connector,
+        previewValues,
+      );
```

Comment:
It's not good that this exists in two places (here and in `submitAddSourceForm`). It seems like the form data is processed differently for the preview than it is for the form submission.

Could we keep _one_ source-of-truth for the data processing logic?